src/main/java/uk/gov/hmcts/dts/fact/services/admin/AdminCourtAreasOfLawService.java

```diff
@@ -29,7 +36,40 @@ public AdminCourtAreasOfLawService(final CourtRepository courtRepository) {
             .orElseThrow(() -> new NotFoundException(slug));
     }
 
+    @Transactional(rollbackFor = {RuntimeException.class})
+    public List<AreaOfLaw> updateAreasOfLawForCourt(final String slug, final List<AreaOfLaw> areasOfLaw) {
+        final Court courtEntity = courtRepository.findBySlug(slug)
+            .orElseThrow(() -> new NotFoundException(slug));
+        List<uk.gov.hmcts.dts.fact.entity.AreaOfLaw> newAreasOfLawList = getNewAreasOfLaw(areasOfLaw);
+        List<CourtAreaOfLaw> newCourtAreasOfLawList = getNewCourtAreasOfLaw(courtEntity, newAreasOfLawList);
+
+        courtAreaOfLawRepository.deleteCourtAreaOfLawByCourtId(courtEntity.getId());
+
+        return courtAreaOfLawRepository
+            .saveAll(newCourtAreasOfLawList)
+            .stream()
+            .map(CourtAreaOfLaw::getAreaOfLaw)
+            .map(AreaOfLaw::new)
+            .collect(toList());
+    }
+
+    private List<uk.gov.hmcts.dts.fact.entity.AreaOfLaw> getNewAreasOfLaw(final List<AreaOfLaw> areaOfLaw) {
+        return areaOfLaw.stream()
+            .map(e -> new uk.gov.hmcts.dts.fact.entity.AreaOfLaw(e.getId(), e.getName()
+            )).collect(toList());
+    }
+
+    private List<CourtAreaOfLaw> getNewCourtAreasOfLaw(final Court court, final List<uk.gov.hmcts.dts.fact.entity.AreaOfLaw> areasOfLaw) {
+        final List<CourtAreaOfLaw> newCourtAreasOfLawList = new ArrayList<>();
+        for (int i = 0; i < areasOfLaw.size(); i++) {
+            newCourtAreasOfLawList.add(new CourtAreaOfLaw(areasOfLaw.get(i), court));
+        }
+        return newCourtAreasOfLawList;
```

Comment:
Nothing wrong with this method here. An alternative is to use stream API in one line:

return areasofLaw.stream().map(a -> new CourtAreaOfLaw(a, court).collect(toList());